src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="columnsDerivedFromSourceEntity">Set of columns in this entity whose values should be
+        /// derived from insertion in the source entity (i.e. source entity would have been the referenced entity).</param>
+        /// <param name="columnsToDerive">Set of columns in this entity whose values are to be
+        /// derived from insertion in the this entity or its subsequent referenced entities and returned to the source entity
+        /// so as to provide values for the corresponding referencing fields (i.e. source entity would have been the referencing entity).</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request.</param>
+        /// <param name="parentEntityName">Parent entity's name.</param>
+        /// <param name="sqlMetadataProviderFactory">Metadata provider factory.</param>
+        private static void ValidateObjectFieldNodes(
+            InputObjectType schemaObject,
+            string entityName,
+            IMiddlewareContext context,
+            IReadOnlyList<ObjectFieldNode> objectFieldNodes,
+            RuntimeConfig runtimeConfig,
+            HashSet<string> columnsDerivedFromSourceEntity,
+            HashSet<string> columnsToDerive,
+            int nestingLevel,
+            string parentEntityName,
+            IMetadataProviderFactory sqlMetadataProviderFactory)
+        {
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
+            ISqlMetadataProvider metadataProvider = sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            SourceDefinition sourceDefinition = metadataProvider.GetSourceDefinition(entityName);
+            Dictionary<string, IValueNode?> backingColumnData = MultipleCreateOrderHelper.GetBackingColumnDataFromFields(context, entityName, objectFieldNodes, metadataProvider);
+
+            // Set of columns in the current entity whose values can be derived via:
+            // a. User input
+            // b. Insertion in the source referenced entity (if current entity is a referencing entity for a relationship with its parent entity)
+            // c. Insertion in the target referenced entity (if the current entity is a referencing entity for a relationship with its target entity)
+            //
+            // derivableColumnsFromRequestBody is initialized with the set of columns whose value is specified in the input (a).
+            Dictionary<string, string> derivableColumnsFromRequestBody = new();
+            foreach ((string backingColumnName, _) in backingColumnData)
+            {
+                derivableColumnsFromRequestBody.TryAdd(backingColumnName, entityName);
+            }
+
+            // When the parent entity is a referenced entity in a relationship, the values of the referencing columns
```

Comment:
How can we be sure at this point that the parent entity is the referenced entity?

This validation runs when the parent entity is the referencing as well as the referenced entity right? Should this be run after making sure that parent entity is the referenced entity?